File: javascript/004/004.js
"use strict";

let palindromes = [];

for (let x = 100; x < 1000; x += 1) {
  for (let y = 100; y < 1000; y += 1) {
    const product = x * y;

    const s = String(product);

    if (s === s.split("").reverse().join("")) {
      palindromes.push(product);
    }
  }
}

console.log(Math.max.apply(null, palindromes));

File: rust/004/004.rs
fn main() {
    let mut x: u64;

    let mut y: u64;

    let mut palindromes = vec![];

    x = 100u64;

    while x <= 999u64 {
        y = 100u64;

        while y <= 999u64 {
            let product: u64 = x * y;

            let s = product.to_string();

            if s == s.chars().rev().collect::<String>() {
                palindromes.push(product);
            }

            y += 1u64;
        }

        x += 1u64;
    }

    let mut max = palindromes[0];

    let mut i = 1usize;

    while i < palindromes.len() {
        if palindromes[i] > max {
            max = palindromes[i];
        }

        i += 1usize;
    }

    println!("{}", max);
}
